database/index.go

```diff
@@ -48,6 +48,53 @@ func term(s string) string {
 
 var httpPat = regexp.MustCompile(`https?://\S+`)
 
+func collectSynopsisTerms(terms map[string]bool, synopsis string) {
+
+	synopsis = httpPat.ReplaceAllLiteralString(synopsis, "")
+
+	fields := strings.FieldsFunc(synopsis, isTermSep)
+	for i := range fields {
+		fields[i] = strings.ToLower(fields[i])
+	}
+
+	if len(fields) > 0 && fields[0] == "the" {
+		fields = fields[1:]
+	}
+
+	// Ignore boilerplate in the following common patterns:
+	//      Package foo ...
+	//      Package foo implements ...
+	//      Package foo provides ...
+	//      Package foo contains ...
+	//      Command foo ...
+	if len(fields) > 0 {
```

Comment:
If you've just trimmed off the leading "the", the following won't match strings like "The foo package implements..." Is that intentional? Are such strings uncommon enough that it doesn't matter? Or should we handle them too?